.github/workflows/wasmcloud.yml

```diff
@@ -81,6 +81,40 @@ jobs:
         name: wasmcloud-universal-darwin
         path: wasmcloud-universal-darwin
 
+  oci:
```

Comment:
Can we gate this whole job instead of only some of the steps below?

Reply:
We can, but we should not - we should verify that the build works on every build

Reply:
Okay, how about building/running the image in one job always, and uploading an artifact we download in a push job?

The separate `if` checks on the steps makes this job difficult to follow

Reply:
I'll refactor the step into a multiline script, but it will only get more "difficult to follow" as we add pushing of tags and whatnot.
That's just how GitHub Actions work and there's nothing we can do about that.
What you are suggesting would not work